### PR TITLE
Split Aspire.Hosting.Azure into sub-packages.

### PR DESCRIPTION
**Pull request by mitchdenny**
Splitting `Aspire.Hosting.Azure` into sub-packages (e.g. `Aspire.Hosting.Azure.Redis`. The package suffix will be dictated by the corresponding package name in the `Azure.ResourceManager.*` packages. This PR is just the structural change.

- [x] Redis
- [x] AppConfig
- [x] Storage?
- [x] Cosmos
- [x] AppInsights
- [x] LogAnalytics
- [x] OpenAI
- [x] Postgres
- [x] Search
- [x] ServiceBus
- [x] SignalR
- [x] SqlServer

Some resources will stay in the main `Aspire.Hosting.Azure` package. At this point its the provisioning machinery, Bicep resource, construct resoure, KeyVault.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3062)

### CODE REVIEW

**Comment by radical** [comment]
It would be useful to add tests that exercise these nugets in `Aspire.EndToEnd.Tests` - since these tests use the nugets produced in `artifacts` to build, and run a test project.

**Comment by davidfowl** [comment]
> It would be useful to add tests that exercise these nugets in Aspire.EndToEnd.Tests - since these tests use the nugets produced in artifacts to build, and run a test project.

We don't have a reasonable way to test against azure resources yet (AFAIK)

**Comment by radical** [comment]
> > It would be useful to add tests that exercise these nugets in Aspire.EndToEnd.Tests - since these tests use the nugets produced in artifacts to build, and run a test project.
> 
> 
> 
> We don't have a reasonable way to test against azure resources yet (AFAIK)

Ah, yes, Azure!

File: src/Aspire.Hosting.Azure.Storage/Aspire.Hosting.Azure.Storage.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <PackageTags>aspire hosting azure</PackageTags>
+    <Description>Azure resource types for .NET Aspire.</Description>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Aspire.Hosting.Azure\Aspire.Hosting.Azure.csproj" />
+    <PackageReference Include="Azure.Provisioning" />
+    <PackageReference Include="Azure.ResourceManager.Authorization" />
+    <PackageReference Include="Azure.Identity" />
```
Comment (davidfowl) [inline src/Aspire.Hosting.Azure.Storage/Aspire.Hosting.Azure.Storage.csproj:15]: ?

**Review by davidfowl — APPROVED** [review]

File: Aspire.sln [hunk 2]
```diff
@@ -381,6 +381,32 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Seq", "src\A
...
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.PostgreSql", "src\Aspire.Hosting.Azure.PostgreSql\Aspire.Hosting.Azure.PostgreSql.csproj", "{B6F2387B-2AB0-41F4-A609-9EA13E408145}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.Search", "src\Aspire.Hosting.Azure.Search\Aspire.Hosting.Azure.Search.csproj", "{927F564C-D037-4296-AB8F-A7F9FF24B3BB}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.ServiceBus", "src\Aspire.Hosting.Azure.ServiceBus\Aspire.Hosting.Azure.ServiceBus.csproj", "{936E1833-B4B9-4610-98BE-0A6650A36E79}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.SignalR", "src\Aspire.Hosting.Azure.SignalR\Aspire.Hosting.Azure.SignalR.csproj", "{CB7CAE39-F041-4B20-A0C4-D6F44920A647}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.Sql", "src\Aspire.Hosting.Azure.Sql\Aspire.Hosting.Azure.Sql.csproj", "{9FF853BD-FA56-4DA5-A50A-9867F2FAB1F0}"
```
Comment (davidfowl) [inline Aspire.sln:406]: We have Aspire.Hosting.SqlServer. Sql feels a little generic.

Comment (mitchdenny) [reply Aspire.sln:406]: @tg-msft FYI

File: Aspire.sln [hunk 3]
```diff
@@ -381,6 +381,32 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Seq", "src\A
...
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.AppConfiguration", "src\Aspire.Hosting.Azure.AppConfiguration\Aspire.Hosting.Azure.AppConfiguration.csproj", "{148AD9C0-A987-42B1-9C00-7E018CC36F24}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.ApplicationInsights", "src\Aspire.Hosting.Azure.ApplicationInsights\Aspire.Hosting.Azure.ApplicationInsights.csproj", "{445EE1E4-2A1E-4E65-A16F-74CF0C261A50}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.CosmosDB", "src\Aspire.Hosting.Azure.CosmosDB\Aspire.Hosting.Azure.CosmosDB.csproj", "{AE9EDE9F-AC34-4DE1-8DAB-9716924C3699}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.OperationalInsights", "src\Aspire.Hosting.Azure.OperationalInsights\Aspire.Hosting.Azure.OperationalInsights.csproj", "{8CF39705-FCA7-42F5-B3E3-1FBD9100BAFF}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Aspire.Hosting.Azure.PostgreSql", "src\Aspire.Hosting.Azure.PostgreSql\Aspire.Hosting.Azure.PostgreSql.csproj", "{B6F2387B-2AB0-41F4-A609-9EA13E408145}"
```
[the remark this answers is not in the document]
Comment (mitchdenny) [reply Aspire.sln:398]: @tg-msft FYI

File: Aspire.sln (hunk 2 above)
Comment (JoshLove-msft) [reply Aspire.sln:406]: The RM libraries use Sql as well.

**Pull request merged**